osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -180,22 +203,62 @@ public void RemoveAt(int index)
             bindable.UnbindAll();
             startTimeBindables.Remove(hitObject);
 
-            // must be run after any change to hitobject ordering
+            if (isBatchApplying)
+                batchPendingDeletes.Add(hitObject);
+            else
+            {
+                // must be run after any change to hitobject ordering
+                beatmapProcessor?.PreProcess();
+                processHitObject(hitObject);
+                beatmapProcessor?.PostProcess();
+
+                HitObjectRemoved?.Invoke(hitObject);
+            }
+        }
+
+        private readonly List<HitObject> batchPendingInserts = new List<HitObject>();
+
+        private readonly List<HitObject> batchPendingDeletes = new List<HitObject>();
+
+        /// <summary>
+        /// Apply a batch of operations in one go, without performing Pre/Postprocessing each time.
+        /// </summary>
+        /// <param name="applyFunction">The function which will apply the batch changes.</param>
+        public void ApplyBatchChanges(Action<EditorBeatmap> applyFunction)
+        {
+            if (isBatchApplying)
+                throw new InvalidOperationException("Attempting to perform a batch application from within an existing batch");
+
+            isBatchApplying = true;
+
+            applyFunction(this);
+
             beatmapProcessor?.PreProcess();
-            processHitObject(hitObject);
             beatmapProcessor?.PostProcess();
```

Comment:
It looks to me like `PostProcess` ought to be called after processing the batch inserts and deletes, right? Can't seem to find a way to break it, though...

Reply:
Ah, nice catch. I think this won't be visible in our current rulesets as we don't use this flow, but does need to happen that way around I believe.